app/controllers/results_controller.rb

```diff
@@ -166,11 +160,22 @@ def show
                                  peer_visible: is_review)
           criteria_info = criteria.pluck_to_hash(*fields)
           marks_info = criteria.joins(:marks)
-                               .where('marks.result_id': result.id)
-                               .pluck_to_hash(*fields, 'marks.mark')
-                               .group_by { |h| h[:id] }
-          criteria_info.map do |h|
-            info = marks_info[h[:id]]&.first || h.merge('marks.mark': nil)
+                              .where('marks.result_id': result.id)
+                              .pluck_to_hash(*fields, 'marks.mark')
+                              .group_by { |h| h[:id] }
+          # adds an extra levels field to the marks info hash with the same rubric criterion id
+          levels_info = []
+          if klass == RubricCriterion
```

Comment:
Move this check to inside the `criteria_info.map` loop (this avoids having to repeat the loop structure)